### DISCUSSION TITLE
Let's list features for the Discord bot

### DISCUSSION THREAD

**Discussion by heypoom** [kaogeek/kaogeek-discord-bot#2] [Ideas]
มาลองลิสต์ feature ใน Discord Bot กัน!

**Comment by ibsfb** [comment]
1. หน้าแนะนำ ไกด์ไลน์คำสั่ง
2. ติดตามนโยบาย
3. Search keywords นโยบายว่าตัวเองคิดไอเดียซ้ำไหม
4. หาพาร์ทเนอร์
5. FAQs
6. เก็บ score card, up lvl

**Comment by LagSeeN** [comment]
ระบบ Ticket แจ้งปัญหาต่าง ๆ

**Comment by swissnp** [comment]
skill/interest roles

**Comment by Tapao-NonSen** [comment]
Verification สำหรับยืนยันก่อนเข้าเซิฟเวอร์กันสเเปมอีกชั้นครับ https://github.com/nates/ward

![68747470733a2f2f692e696d6775722e636f6d2f746d7263796a462e706e67](https://github.com/creatorsgarten/kaogeek-discord-bot/assets/40026698/423cf430-83a7-411d-909c-addb375a6191)

**Reply by Zimbo333** [reply]
เห็นด้วยมากฮะ กันสแปม กันบอท อย่างน้อยให้เค้ารู้ตัวว่าถ้าทำผิดหรือสแปม ก็รู้ข้อมูลของคุณคร่าวๆ นะ แต่อาจจะต้องดูว่าจะเก็บข้อมูลอะไรมั้ย แล้วก็จะยืนยันด้วยอะไรบ้าง

แอบสนใจเบอร์โทรครับ

**Reply by anwam** [reply]
ถ้าป้องกัน Spam เฉยๆ Cloudflare Turnstile เพียงพอมั้ยครับ?
หรือเราต้องการ verify identity ด้วย

**Comment by opecgame** [comment]
Report Abuse User
- > Interaction Rigth clivk แล้วจะไปแจ้งให้แอดมินตรวจสอบอีกทีนึง

**Comment by themistymoon** [comment]
/progress {name} ใช้ list เพื่อแสดง list progress ต่างๆ เช่น นโยบาย

**Comment by Tee35973** [comment]
ไกด์คำส่ง
ระบบTicket
Manual moderation command
Welcome message(?)
ออโต้มอด
ระบบประกาศบางประเภท(Embed)จะได้ดู official และสามรถเชื่อมwebhookเพื่อประกาศข่าวหรือนโยบายจาก externalได้

**Comment by kang49** [comment]
แสดง top10 forum lists จัดตาม like or comments value

**Comment by understann** [comment]
แจ้ง event พรรค

**Comment by MikihinaSann** [comment]
AIChat สำหรับตอบคำถามเกี่ยวกับ ก้าวไกล อธิเช่นแบบ
User: ก้าวไกลจะทำอะไรในอนาคตบ้าง
BOT: ก็จะตอบออกมาอะไรแบบนี้ครับ

ถ้าเป็นคำถามที่ไม่เคยเจออาจจะส่งกลับมาหลังบ้านเพื่อ Train อะไรงี้ครับ

**Comment by Tee35973** [comment]
Verify เหมือนเป็นapllication แล้วให้คนมาคอยอ่านมั้ยครับเพื่อความชัวร์ ไม่ก็ต้องเข้าก่อนสักประมาณ 1 ชั่งโมง/ 1 วันเพื่อที่จะverify

**Reply by Tee35973** [reply]
เพื่อความแน่ใจอีกขั้น

**Comment by narze** [comment]
รวมชื่อกระทู้ไอเดียต่างๆ ใน server ก่อนหน้านี้คือต้องไล่ scrape เอา

<img width="2431" alt="image" src="https://github.com/creatorsgarten/kaogeek-discord-bot/assets/248741/f6bc0b1f-e13a-4d59-b23f-0808a21c8b3e">

**Reply by narze** [reply]
ถ้ามันดึง reactions count ได้ด้วยก็ดี

**Reply by nutnnut** [reply]
ผมว่าระดมไอเดียแบบนี้ bias มากเลยครับ
- เวลาในการโพสมีผลกับคนที่เห็น
- feedback loop (คอมเม้นแล้วเด้งไปอยู่บนสุด)
- ความยากในเนื้อหา (ถ้าเป็นเรื่อง technical คนจะไม่ค่อย engage)

อยากให้มีวิธีเสนอไอเดียที่ normalize/sort ความน่าสนใจที่ดีกว่านับ engagement/reaction ครับ อาจจะต้องมีการกด dislike สแปมด้วย

**Comment by sagelga** [comment]
ขอเป็นอะไรก็ได้ที่จะหาการสนับสนุนเงินทุนพรรคก็ดีนะฮะ เงินไม่มางานไม่เดิน
ref: https://donations.moveforwardparty.org/

**Reply by narze** [reply]
ระวังเคส `เงินบริจาค = รายได้` ด้วยนะครับ 🫠

**Reply by sagelga** [reply]
มันคงไม่โดนหรอกครับ แต่ผมว่ารอฝ่ายกฎหมายมาคอนเฟิร์มดีกว่า
แต่มันสำคัญมากนะฮะถ้าไม่มีเงินบริจาคเนี่ย

**Comment by mineniji** [comment]
เสนอให้แยกบอท ให้ทำงานเป็นส่วนๆ จะได้ไม่ spagetti code เกินไป เช่น
- บอทที่แจ้งนโยบาย
- บอท discord management
- บอทยืนยันตัวตน
- อื่นๆ ตามงานที่ควรมี
- ใครที่มีความถนัดหรือมีความรู้เกี่ยวกับบอทด้านไหน จะได้ช่วยกันดูได้

**Comment by pickyzz** [comment]
- open ticket to Admin โดยที่เก็บสามารถ Log ไว้เป็นหลักฐานการพูดคุยย้อนหลังได้
- Create temporary voice channel และ ปิดห้องเมื่อไม่มีคนอยู่ในห้อง

**Comment by Tapao-NonSen** [comment]
การกรองลิ้งโดยใช้ APIs ของ Google Safe Browsing มาช่วยกรองหรือใช้ Database ที่มีเพื่อกรอง

**Comment by narze** [comment]
AI น้อนไข่ต้ม 
<img width="113" alt="image" src="https://github.com/creatorsgarten/kaogeek-discord-bot/assets/248741/97f2b0d5-20a0-4536-be70-bcfa5e675bdd">

**Comment by SaltyAom** [comment]
ถ้ามี timeline/progress/roadmap หรือรายละเอียดของนโยบาลพรรคได้น่าจะดีนะ

บางทีอยากติดตามขั้นตอนการดำเนินงานที่สนใจแต่ไม่รู้ว่าจะไปตามที่ไหน eg. บางทีก็กระจายไปตาม social network ถ้าได้แบบ link tree แต่สำหรับนโยบาลพรรคได้ก็ดีนะ ยิ่งถ้าดูใน discord ได้ก็น่าจะสะดวกดี eg. อาจจะ generate นโยบาลเป็น forum แล้ว update แล้วให้คนก็ feedback ได้ หรือให้คนช่วยดูได้ จะได้โปร่งใสแล้วมี log ว่าทำงานจริง

**Comment by tnwisme** [comment]
อัพเดตโพสต์หรือทวีตของพรรคผ่านบอท

**Comment by HelloLazyWorld** [comment]
JOB FINDER นี่ล่ะ โปรไฟล์ชาว dev กำลังมานินะ

**Comment by understann** [comment]
Create profile ตัวเองใส่ข้อมูลตัวเอง แล้ว render มาเป็นเหมือนบัตร ประจำตัว

**Comment by freezeza007** [comment]
เสนอเรื่อง #Role
เห็นว่าดิสคอร์ดมีจำกัดจำนวนคนในแชลแนลน่ะครับ สุดท้ายแล้วก็ไม่สามารถรับคนมากกว่าที่จะรับได้ 
หากจะทำโรลโดยมีหลักการพื้นที่ เช่นแยกเป็นภาค ใต้ กลาง เหนือ อีสาน... สุดท้ายผมว่าก็ต้องมีระบบตัวแทนคล้ายกับสส.แต่อันนี้แทนในฐานะ เทค-เดฟ จากพื้นที่ต่างๆ โดยมีข้อดีนอกเหนือจากการโค๊ดดิ่ง คือ หากวันใดพรรคหรือผู้สมัคร มีความต้องการจะสอบถามปัญหาอะไร ในตัวพื้นที่นั้นๆ จะได้มีคอนเทค กันตามโรลที่ตั้งไว้ด้วย เช่น บางจังหวัด พื้นที่นึงมีปัญหาเกิดขึ้น หากเรามี เทค-เดฟ ซึ่งอยู่ในพื้นที่ มากกว่า1 คนจะช่วยให้เราสามารถพูดคุยกันถึงปัญหาที่เกิดขึ้นในเชิงนโยบายของพรรคในพื้นที่ดังกล่าวนั้นด้วย อาจจะมีการคัด เลือก หรือตั้งเงื่อนไข ให้เหมาะกับความสามารถของคนๆนั้น ตามระดับเลเวลก็ได้ครับ เพื่อประโยชน์ในการแก้ไขปัญหาของบ้านเมืองอย่างรวดเร็วในยุคนี้ครับ

**Comment by Rindarmare** [comment]
https://www.figma.com/file/d22o2ke1JNrRSWtBho2l5F/Kao-Geek?type=whiteboard&node-id=0-1

**Comment by PusanStudio** [comment]
เพิ่ม lockdown mode เวลาโดน raid 
จะได้ให้บอท
- ปิดinvite 
- slowndown chat 
- ปิด voice connection

**Comment by damp11113** [comment]
Report user และ ระบบ verify

**Comment by ghost** [comment]
/ask chatgpt ด้วย chatgpt api 

ประมาณ https://github.com/biscxit/wumpus-gpt
